mediator/med_phases_prep_ocn_mod.F90

```diff
@@ -632,202 +628,27 @@ subroutine med_phases_prep_ocn_custom_nems(gcomp, rc)
-    customwgt(:) = ofrac(:) * (1.0 - 0.06)
-    call med_merge_field(is_local%wrap%FBExp(compocn),      'Foxx_swnet_vdr', &
-         FBinA=is_local%wrap%FBImp(compatm,compocn), fnameA='Faxa_swvdr'    , wgtA=customwgt, &
-         FBinB=is_local%wrap%FBImp(compice,compocn), fnameB='Fioi_swpen_vdr', wgtB=ifrac, rc=rc)
-    if (ChkErr(rc,__LINE__,u_FILE_u)) return
-    call med_merge_field(is_local%wrap%FBExp(compocn),      'Foxx_swnet_vdf', &
-         FBinA=is_local%wrap%FBImp(compatm,compocn), fnameA='Faxa_swvdf'    , wgtA=customwgt, &
-         FBinB=is_local%wrap%FBImp(compice,compocn), fnameB='Fioi_swpen_vdf', wgtB=ifrac, rc=rc)
-    if (ChkErr(rc,__LINE__,u_FILE_u)) return
-    call med_merge_field(is_local%wrap%FBExp(compocn),      'Foxx_swnet_idr', &
-         FBinA=is_local%wrap%FBImp(compatm,compocn), fnameA='Faxa_swndr'    , wgtA=customwgt, &
-         FBinB=is_local%wrap%FBImp(compice,compocn), fnameB='Fioi_swpen_idr', wgtB=ifrac, rc=rc)
-    if (ChkErr(rc,__LINE__,u_FILE_u)) return
-    call med_merge_field(is_local%wrap%FBExp(compocn),      'Foxx_swnet_idf', &
-         FBinA=is_local%wrap%FBImp(compatm,compocn), fnameA='Faxa_swndf'    , wgtA=customwgt, &
-         FBinB=is_local%wrap%FBImp(compice,compocn), fnameB='Fioi_swpen_idf', wgtB=ifrac, rc=rc)
-    if (ChkErr(rc,__LINE__,u_FILE_u)) return
-
-    deallocate(customwgt)
-
-    if (dbug_flag > 20) then
-       call ESMF_LogWrite(trim(subname)//": done", ESMF_LOGMSG_INFO)
-    end if
-    call t_stopf('MED:'//subname)
-
-  end subroutine med_phases_prep_ocn_custom_nems
-
-  !-----------------------------------------------------------------------------
-  subroutine med_phases_prep_ocn_custom_nemsdata(gcomp, rc)
-
-    ! ----------------------------------------------
-    ! Custom calculation for nems_orig_data
-    ! ----------------------------------------------
-
-    use ESMF , only : ESMF_GridComp
-    use ESMF , only : ESMF_LogWrite, ESMF_LOGMSG_INFO, ESMF_SUCCESS
-    use ESMF , only : ESMF_FAILURE,  ESMF_LOGMSG_ERROR
-
-    ! input/output variables
-    type(ESMF_GridComp)  :: gcomp
-    integer, intent(out) :: rc
-
-    ! local variables
-    type(InternalState) :: is_local
-    real(R8), pointer   :: ocnwgt1(:)   ! NEMS_orig_data
-    real(R8), pointer   :: icewgt1(:)   ! NEMS_orig_data
-    real(R8), pointer   :: wgtp01(:)    ! NEMS_orig_data
-    real(R8), pointer   :: wgtm01(:)    ! NEMS_orig_data
-    real(R8), pointer   :: customwgt(:) ! NEMS_orig_data
-    real(R8), pointer   :: ifrac(:)
-    real(R8), pointer   :: ofrac(:)
-    integer             :: lsize
-    integer             :: n
-    real(R8)        , parameter    :: const_lhvap = 2.501e6_R8  ! latent heat of evaporation ~ J/kg
-    character(len=*), parameter    :: subname='(med_phases_prep_ocn_custom_nemsdata)'
-    !---------------------------------------
+    if (trim(coupling_mode) == 'nems_orig' .or. trim(coupling_mode) == 'nems_frac') then
```

Comment:
Does this mean that for datm, the coupled fields inside this if statement are already set up and ready for ocean? Also, is there a mask defined on atm export grid(or ocean import field grid) to compute atm and ocean fluxes on open water points only? It seems to me the flux computation here covers both open water and ice points, would the fluxes on fractional ice points (part ice +part open water) be merged from bulk formula computation and ice model? I seem not able to find the code to do that.

Reply:
For the first part of your question, yes these do not need to be calculated in prep-ocn because they're able to be done via the auto-merge in esmFldsExchange: [LN 259](https://github.com/DeniseWorthen/CMEPS/blob/af0d961386080ced7f63a14ca194b4c2bbe5a95d/mediator/esmFldsExchange_nems_mod.F90#L259). This is where you're "merging" (really just weighting by ofrac) for example the sensible heat flux calculated by the compmed. 

The CMEPS aoflux calculation is the same aoflux calculation done by the NEMS mediator, using the atm import state. When we switched to bulk formulas for the NEMS mediator, it is equivalent to using just ofrac weighting. I was able to reproduce the current DATM-MOM6-CICE5 baseline with a NEMS mediator [using only ofrac](https://github.com/DeniseWorthen/NEMS/blob/ea1558ad219150465e42c2e6d95982665f9e5b80/src/module_MEDIATOR.F90#L5478). In this case, all the ocean fluxes are coming from the aoflux calculation, weighted by ofrac. 

Some fields are sent separately to MOM6 from ice and atm but that is how MOM6 expects them so that it can do it's budgeting.

Reply:
Thanks for the explanation, the code changes look good.